api_client/python/tools/timesketch_importer.py

```diff
@@ -0,0 +1,269 @@
+# Copyright 2020 Google Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""A simple frontend to the Timesketch data importer.
+
+This tool is limited to upload files to Timesketch, which is only
+part of the capabilities of the importer API.
+"""
+from __future__ import unicode_literals
+
+import argparse
+import getpass
+import logging
+import os
+import sys
+
+from typing import Dict
+
+import yaml
+
+from timesketch_api_client import client
+from timesketch_api_client import importer
+from timesketch_api_client import sketch
+
+
+logging.basicConfig(level=os.environ.get('LOGLEVEL', 'INFO'))
+
+
+def get_client(
+        host: str, username: str, password: str = '', client_id: str = '',
+        client_secret: str = '', run_local: bool = False
+        ) -> client.TimesketchApi:
+    """Returns a Timesketch API client.
+
+    Args:
+        host (str): the Timesketch host.
+        username (str): the username to authenticate with.
+        password (str): optional used if OAUTH is not the authentication
+                        mechanism.
+        client_id (str): if OAUTH is used then a client ID needs to be set.
+        client_secret (str): if OAUTH is used then a client secret needs to be
+                             set.
+        run_local (bool): if OAUTH is used to authenticate and set to True
+                          then the authentication URL is printed on screen
+                          instead of starting a web server, this suits well
+                          if the connection is over a SSH connection for
+                          instance.
+
+    Raises:
+        TypeError: If a non supported authentication mode is passed in.
+
+    Returns:
+        A Timesketch API client object.
+    """
+    if run_local and client_secret:
+        auth_mode = 'oauth_local'
+    elif client_secret:
+        auth_mode = 'oauth'
+    elif password:
+        auth_mode = 'timesketch'
+    else:
+        raise TypeError(
+            'Neither password nor client secret provided, unable '
+            'to authenticate')
+
+    if not host.startswith('http'):
+        host = 'https://{0:s}'.format(host)
+
+    api_client = client.TimesketchApi(
+        host_uri=host, username=username, password=password,
+        client_id=client_id, client_secret=client_secret, auth_mode=auth_mode)
+
+    return api_client
+
+
+def upload_file(
```

Comment:
Do we want to validate the file extension (.plaso, csv, jsonl) here and exit early? I know you are checking this in the client but maybe worth doing here?

Reply:
we could do that... done